extralit-server/src/extralit_server/jobs/ocr_jobs.py

```diff
@@ -12,3 +12,243 @@
+            raise
+
+        # Extract bounding boxes using our utility functions
+        tables = extract_table_bboxes(layout_result)
+        figures = extract_figure_bboxes(layout_result)
+        text_blocks = extract_text_bboxes(layout_result) if extract_text else []
+
+        result = {
+            "tables": tables,
+            "figures": figures,
+            "text_blocks": text_blocks,
+            "metadata": {
+                "source": "marker",
+                "pdf_path": str(pdf_path),
+                "pages_processed": pages or "all",
+                "total_elements": len(tables) + len(figures) + len(text_blocks),
+                "extract_text": extract_text,
+                "processing_time": None,
+            },
+        }
+
+        # Update job metadata with results
+        # current_job.meta.update(
+        #     {
+        #         "layout_extraction_complete": True,
+        #         "tables_found": len(tables),
+        #         "figures_found": len(figures),
+        #         "text_blocks_found": len(text_blocks),
+        #     }
+        # )
+        # current_job.save_meta()
+
+        _LOGGER.info(f"Marker layout extraction completed. Found {len(tables)} tables, {len(figures)} figures")
+        return result
+
+    except Exception as e:
+        _LOGGER.error(f"Error in marker layout extraction job: {e}", exc_info=True)
+        # current_job.meta["error"] = str(e)
+        # current_job.save_meta()
+        raise
+
+
+def _call_marker_layout_detection(pdf_path: str, pages: Optional[str] = None) -> dict[str, Any]:
+    """
+    Call Marker's layout detection API using the standard PdfConverter.
+
+    Args:
+        pdf_path: Path to the PDF file
+        pages: Optional list of page numbers to process (0-indexed)
+
+    Returns:
+        Marker's layout detection results with block structure
+    """
+    # Basic input validation
+    if not pdf_path:
+        raise ValueError("PDF path cannot be empty")
+
+    pdf_file = Path(pdf_path)
+    if not pdf_file.exists():
+        raise FileNotFoundError(f"PDF file does not exist: {pdf_path}")
+
+    if pdf_file.suffix.lower() != ".pdf":
+        raise ValueError(f"File is not a PDF: {pdf_path}")
+
+    try:
+        # Create optimized configuration for layout detection
+        config_dict = {
+            "output_format": "json",
+            "parallel_factor": 1,
+        }
+
+        if pages is not None:
+            config_dict["page_range"] = pages
+
+        config = ConfigParser(config_dict)
+        model_dict = create_model_dict()
+
+        converter = PdfConverter(
+            config=config.generate_config_dict(),
+            artifact_dict=model_dict,
+        )
+
+        # Convert PDF - this will return a Document object with detected layout
+        result: "MarkdownOutput | JSONOutput" = converter(pdf_path)  # noqa: UP037
+        print(type(result))
+        pprint(result.model_dump())
+
```

Comment:
@priyankeshh In order to make `marker` not use the OCR models, you will have to edit the model_dict and config_dict here